distributed/client.py

```diff
@@ -727,10 +732,42 @@ def __init__(
 
         ReplayExceptionClient(self)
 
+    @contextmanager
+    def as_current(self):
+        """Thread-local, Task-local context manager that causes the Client.current class
+        method to return self. This is used when a method of Client needs to propagate a
+        reference to self deep into the stack through generic methods that shouldn't be
+        aware of this class.
```

Comment:
Since this only sets the contextvar (and thus only works in an async context), I think this method may be confusing for most users. Perhaps we should keep this as a private method for now (i.e. move to `_as_current(self)`)?

Reply:
The big use case for a final user is when he pickles a dask collection that internally contains futures, e.g. to dump it to disk for forensics purposes, and then wants to unpickle it and he has more than one Client open. I updated the description to reflect this use case, see if you like it.